Missing dependent libraries of embulk-core 0.9.0

I tried to upgrade my plugin to v0.9.0 but encountered the error below.

```sh
sishimura% ./gradlew compileKotlin
:compileKotlin
Using kotlin incremental compilation
w: The '-d' option with a directory destination is ignored because '-module' is specified
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/RemoteFileInputPlugin.kt: (3, 12): Unresolved reference: fasterxml
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/RemoteFileInputPlugin.kt: (4, 12): Unresolved reference: google
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/RemoteFileInputPlugin.kt: (33, 27): Unresolved reference: Optional
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/RemoteFileInputPlugin.kt: (49, 26): Unresolved reference: Optional
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/RemoteFileInputPlugin.kt: (75, 19): Unresolved reference: Optional
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/RemoteFileInputPlugin.kt: (79, 22): Unresolved reference: Optional
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/RemoteFileInputPlugin.kt: (83, 23): Unresolved reference: Optional
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/RemoteFileInputPlugin.kt: (153, 25): Unresolved reference: it
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/RemoteFileInputPlugin.kt: (158, 57): Unresolved reference: it
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/RemoteFileInputPlugin.kt: (202, 14): Unresolved reference: JsonProperty
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/RemoteFileInputPlugin.kt: (203, 14): Unresolved reference: JsonProperty
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/RemoteFileInputPlugin.kt: (204, 14): Unresolved reference: JsonProperty
e: /path/of/embulk-input-remote/src/main/kotlin/org/embulk/input/remote/SSHClient.kt: (38, 48): Unresolved reference: it
:compileKotlin FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1.716 secs
```

It seems that dependent libraries of embulk-core 0.9.0 cannot be resolved.

```sh
sishimura% ./gradlew dependencies --configuration compile
:dependencies

------------------------------------------------------------
Root project
------------------------------------------------------------

compile - Dependencies for source set 'main'.
+--- org.jetbrains.kotlin:kotlin-stdlib:1.1.1
|    \--- org.jetbrains:annotations:13.0
+--- com.hierynomus:sshj:0.19.1
|    +--- org.slf4j:slf4j-api:1.7.7
|    +--- org.bouncycastle:bcprov-jdk15on:1.51
|    +--- org.bouncycastle:bcpkix-jdk15on:1.51
|    |    \--- org.bouncycastle:bcprov-jdk15on:1.51
|    +--- com.jcraft:jzlib:1.1.3
|    \--- net.i2p.crypto:eddsa:0.1.0
+--- com.jcraft:jzlib:1.1.3
\--- org.embulk:embulk-core:0.9.0

(*) - dependencies omitted (listed previously)

BUILD SUCCESSFUL

Total time: 0.729 secs
```

They can be resolved on embulk-core 0.8.39

```sh
sishimura% ./gradlew dependencies --configuration compile 
:dependencies

------------------------------------------------------------
Root project
------------------------------------------------------------

compile - Dependencies for source set 'main'.
+--- org.jetbrains.kotlin:kotlin-stdlib:1.1.1
|    \--- org.jetbrains:annotations:13.0
+--- com.hierynomus:sshj:0.19.1
|    +--- org.slf4j:slf4j-api:1.7.7 -> 1.7.12
|    +--- org.bouncycastle:bcprov-jdk15on:1.51
|    +--- org.bouncycastle:bcpkix-jdk15on:1.51
|    |    \--- org.bouncycastle:bcprov-jdk15on:1.51
|    +--- com.jcraft:jzlib:1.1.3
|    \--- net.i2p.crypto:eddsa:0.1.0
+--- com.jcraft:jzlib:1.1.3
\--- org.embulk:embulk-core:0.8.39
     +--- org.embulk:embulk-jruby-strptime:0.8.39
     |    \--- org.jruby:jruby-complete:9.1.13.0
     +--- org.embulk:guice-bootstrap:0.1.1
     |    \--- com.google.inject:guice:4.0
     |         +--- javax.inject:javax.inject:1
     |         +--- aopalliance:aopalliance:1.0
     |         \--- com.google.guava:guava:16.0.1 -> 18.0
     +--- com.google.guava:guava:18.0
     +--- com.google.inject:guice:4.0 (*)
     +--- com.google.inject.extensions:guice-multibindings:4.0
     |    \--- com.google.inject:guice:4.0 (*)
     +--- javax.inject:javax.inject:1
     +--- com.fasterxml.jackson.core:jackson-annotations:2.6.7
     +--- com.fasterxml.jackson.core:jackson-core:2.6.7
     +--- com.fasterxml.jackson.core:jackson-databind:2.6.7
     |    +--- com.fasterxml.jackson.core:jackson-annotations:2.6.0 -> 2.6.7
     |    \--- com.fasterxml.jackson.core:jackson-core:2.6.7
     +--- com.fasterxml.jackson.datatype:jackson-datatype-guava:2.6.7
     |    +--- com.fasterxml.jackson.core:jackson-databind:2.6.7 (*)
     |    +--- com.fasterxml.jackson.core:jackson-core:2.6.7
     |    \--- com.google.guava:guava:15.0 -> 18.0
     +--- com.fasterxml.jackson.datatype:jackson-datatype-joda:2.6.7
     |    +--- com.fasterxml.jackson.core:jackson-annotations:2.6.0 -> 2.6.7
     |    +--- com.fasterxml.jackson.core:jackson-core:2.6.7
     |    +--- com.fasterxml.jackson.core:jackson-databind:2.6.7 (*)
     |    \--- joda-time:joda-time:2.2 -> 2.9.2
     +--- com.fasterxml.jackson.module:jackson-module-guice:2.6.7
     |    +--- com.fasterxml.jackson.core:jackson-core:2.6.7
     |    +--- com.fasterxml.jackson.core:jackson-databind:2.6.7 (*)
     |    \--- com.google.inject:guice:3.0 -> 4.0 (*)
     +--- ch.qos.logback:logback-classic:1.1.3
     |    +--- ch.qos.logback:logback-core:1.1.3
     |    \--- org.slf4j:slf4j-api:1.7.7 -> 1.7.12
     +--- org.slf4j:slf4j-api:1.7.12
     +--- org.jruby:jruby-complete:9.1.13.0
     +--- com.google.code.findbugs:annotations:3.0.0
     +--- org.yaml:snakeyaml:1.18
     +--- javax.validation:validation-api:1.1.0.Final
     +--- org.apache.bval:bval-jsr303:0.5
     |    +--- org.apache.bval:bval-core:0.5
     |    |    +--- org.apache.commons:commons-lang3:3.1 -> 3.4
     |    |    \--- commons-beanutils:commons-beanutils-core:1.8.3
     |    \--- org.apache.commons:commons-lang3:3.1 -> 3.4
     +--- io.airlift:slice:0.9
     +--- joda-time:joda-time:2.9.2
     +--- io.netty:netty-buffer:4.0.44.Final
     |    \--- io.netty:netty-common:4.0.44.Final
     +--- org.fusesource.jansi:jansi:1.11
     +--- org.msgpack:msgpack-core:0.8.11
     +--- com.ibm.icu:icu4j:54.1.1
     +--- org.eclipse.aether:aether-api:1.1.0
     +--- org.eclipse.aether:aether-spi:1.1.0
     |    \--- org.eclipse.aether:aether-api:1.1.0
     +--- org.eclipse.aether:aether-util:1.1.0
     |    \--- org.eclipse.aether:aether-api:1.1.0
     +--- org.eclipse.aether:aether-impl:1.1.0
     |    +--- org.eclipse.aether:aether-api:1.1.0
     |    +--- org.eclipse.aether:aether-spi:1.1.0 (*)
     |    \--- org.eclipse.aether:aether-util:1.1.0 (*)
     \--- org.apache.maven:maven-aether-provider:3.3.9
          +--- org.apache.maven:maven-model:3.3.9
          |    +--- org.codehaus.plexus:plexus-utils:3.0.22
          |    \--- org.apache.commons:commons-lang3:3.4
          +--- org.apache.maven:maven-model-builder:3.3.9
          |    +--- org.codehaus.plexus:plexus-utils:3.0.22
          |    +--- org.codehaus.plexus:plexus-interpolation:1.21
          |    +--- org.codehaus.plexus:plexus-component-annotations:1.6
          |    +--- org.apache.maven:maven-model:3.3.9 (*)
          |    +--- org.apache.maven:maven-artifact:3.3.9
          |    |    +--- org.codehaus.plexus:plexus-utils:3.0.22
          |    |    \--- org.apache.commons:commons-lang3:3.4
          |    +--- org.apache.maven:maven-builder-support:3.3.9
          |    |    +--- org.codehaus.plexus:plexus-utils:3.0.22
          |    |    \--- org.apache.commons:commons-lang3:3.4
          |    +--- com.google.guava:guava:18.0
          |    \--- org.apache.commons:commons-lang3:3.4
          +--- org.apache.maven:maven-repository-metadata:3.3.9
          |    \--- org.codehaus.plexus:plexus-utils:3.0.22
          +--- org.eclipse.aether:aether-api:1.0.2.v20150114 -> 1.1.0
          +--- org.eclipse.aether:aether-spi:1.0.2.v20150114 -> 1.1.0 (*)
          +--- org.eclipse.aether:aether-util:1.0.2.v20150114 -> 1.1.0 (*)
          +--- org.eclipse.aether:aether-impl:1.0.2.v20150114 -> 1.1.0 (*)
          +--- org.codehaus.plexus:plexus-component-annotations:1.6
          +--- org.codehaus.plexus:plexus-utils:3.0.22
          \--- org.apache.commons:commons-lang3:3.4

(*) - dependencies omitted (listed previously)

BUILD SUCCESSFUL

Total time: 4.518 secs
```

It happens when creating new plugin.

```sh
sishimura% embulk new java-file-input foo
2018-02-07 20:37:23.612 +0900: Embulk v0.9.0

********************************** INFORMATION **********************************
  Join us! Embulk-announce mailing list is up for IMPORTANT announcement such as
    compatibility-breaking changes and key feature updates.
  https://groups.google.com/forum/#!forum/embulk-announce
*********************************************************************************

Creating embulk-input-foo/

Plugin template is successfully generated.
Next steps:

  $ cd embulk-input-foo
  $ ./gradlew package

sishimura% cd embulk-input-foo
sishimura% ./gradlew package                                                                                                                                                             [embulk-input-foo]

> Task :compileJava
/Users/shinichi/IdeaProjects/embulk-input-foo/src/main/java/org/embulk/input/foo/FooFileInputPlugin.java:6: エラー: パッケージcom.google.common.baseは存在しません
import com.google.common.base.Optional;
                             ^
/Users/shinichi/IdeaProjects/embulk-input-foo/src/main/java/org/embulk/input/foo/FooFileInputPlugin.java:7: エラー: パッケージcom.google.common.collectは存在しません
import com.google.common.collect.ImmutableList;
                                ^
/Users/shinichi/IdeaProjects/embulk-input-foo/src/main/java/org/embulk/input/foo/FooFileInputPlugin.java:41: エラー: シンボルを見つけられません
        public Optional<String> getOption3();
               ^
  シンボル:   クラス Optional
  場所: インタフェース PluginTask
エラー3個


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
1 actionable task: 1 executed
```

--- a/build.gradle
+++ b/build.gradle
@@ -1,10 +1,20 @@
+// TODO: Remove this block once we re-upgrade jruby-gradle-jar-plugin..
+// See #1: https://github.com/embulk/embulk/issues/954
+// See #2: https://github.com/jruby-gradle/jruby-gradle-plugin/issues/297
+repositories {
+    mavenLocal()
+    maven { url 'http://rubygems-proxy.torquebox.org/releases' }
+}
+
 buildscript {
     repositories {
         jcenter()
     }
     dependencies {
         classpath 'com.github.jengelman.gradle.plugins:shadow:2.0.2'
-        classpath 'com.github.jruby-gradle:jruby-gradle-jar-plugin:1.5.0'
+        // TODO: Re-upgrade jruby-gradle-jar-plugin.
+        // See: https://github.com/embulk/embulk/issues/954
+        classpath 'com.github.jruby-gradle:jruby-gradle-jar-plugin:1.0.1'
         classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.7.3'
     }
 }
@@ -118,7 +128,6 @@ configure(subprojects.findAll { ['embulk-core', 'embulk-standards', 'embulk-test
     }
 
     artifacts {
-        archives jar
         archives testsJar
         archives sourcesJar
         archives javadocJar
@@ -127,7 +136,7 @@ configure(subprojects.findAll { ['embulk-core', 'embulk-standards', 'embulk-test
     publishing {
         publications {
             bintrayMavenRelease(MavenPublication) {
-                artifact jar
+                from components.java
                 artifact testsJar
                 artifact sourcesJar
                 artifact javadocJar

--- a/embulk-core/build.gradle
+++ b/embulk-core/build.gradle
@@ -1,3 +1,11 @@
+// TODO: Remove this block once we re-upgrade jruby-gradle-jar-plugin..
+// See #1: https://github.com/embulk/embulk/issues/954
+// See #2: https://github.com/jruby-gradle/jruby-gradle-plugin/issues/297
+repositories {
+    mavenLocal()
+    maven { url 'http://rubygems-proxy.torquebox.org/releases' }
+}
+
 apply plugin: 'com.github.jruby-gradle.jar'
 
 // TODO: Have more details in description and summary.

--- a/embulk-docs/build.gradle
+++ b/embulk-docs/build.gradle
@@ -1,3 +1,11 @@
+// TODO: Remove this block once we re-upgrade jruby-gradle-jar-plugin..
+// See #1: https://github.com/embulk/embulk/issues/954
+// See #2: https://github.com/jruby-gradle/jruby-gradle-plugin/issues/297
+repositories {
+    mavenLocal()
+    maven { url 'http://rubygems-proxy.torquebox.org/releases' }
+}
+
 apply plugin: 'com.github.jruby-gradle.base'
 
 import com.github.jrubygradle.JRubyExec

--- a/embulk-standards/build.gradle
+++ b/embulk-standards/build.gradle
@@ -1,3 +1,11 @@
+// TODO: Remove this block once we re-upgrade jruby-gradle-jar-plugin..
+// See #1: https://github.com/embulk/embulk/issues/954
+// See #2: https://github.com/jruby-gradle/jruby-gradle-plugin/issues/297
+repositories {
+    mavenLocal()
+    maven { url 'http://rubygems-proxy.torquebox.org/releases' }
+}
+
 apply plugin: 'com.github.jruby-gradle.jar'
 
 // TODO: Have more details in description and summary.

